segments/weather.sh

```diff
@@ -118,8 +114,14 @@ __yrno() {
 		fi
 		# condition_symbol=$(__get_yrno_condition_symbol "$condition" "$sunrise" "$sunset")
 		condition_symbol=$(__get_yrno_condition_symbol "$condition")
-		echo "${condition_symbol} ${degree}°$(echo "$TMUX_POWERLINE_SEG_WEATHER_UNIT" | tr '[:lower:]' '[:upper:]')" | tee "${tmp_file}"
+		# Write the <content @ date>, separated by 2 spaces and @, so we can fetch it later on without having to call 'stat'
+		echo "${condition_symbol} ${degree}°$(echo "$TMUX_POWERLINE_SEG_WEATHER_UNIT" | tr '[:lower:]' '[:upper:]')@$(date +%s)" > "$cache_weather_file"
+		__read_file_content "$cache_weather_file"
+		return
```

Comment:
Though we should move both reading and writing of cache to be done inside `run_segment()` to make it general for any weather provider. Right now only reading is done in the `run_segment()`, but we should do the writing there as well.

See the comment with pseudo code in `run_segment()` 😃 


Thus here we should just echo out the final result of this funciton `__yrno()` i.e.

```suggestion
		echo "${condition_symbol} ${degree}°$(echo "$TMUX_POWERLINE_SEG_WEATHER_UNIT" | tr '[:lower:]' '[:upper:]')@$(date +%s)"
		return 0
```